### PR TITLE
Sample app for demonstration of the MFA in express app

### PR DESCRIPTION
**Pull request by 4gust**
## Purpose

* Added a sample app to demonstrate the working of MFA in profile edit scenario in a express app

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?
Adding a new sample app for demonstration of the MFA in profile edit scenario 

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

File: 1-Authentication/7-sign-in-express-mfa/App/views/id.hbs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<h1>Azure AD</h1>
```
Comment (iambmelt) [inline 1-Authentication/7-sign-in-express-mfa/App/views/id.hbs:1]:
```suggestion
<h1>Entra ID</h1>
```

File: 1-Authentication/7-sign-in-express-mfa/App/auth/AuthProvider.js [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+            res,
+            next,
+            authCodeUrlRequestParams,
+            authCodeRequestParams,
+            msalInstance
+        );
+    }
+
+    doesRequireMFA(token) {
+        // Decode the access token payload
```
Comment (iambmelt) [inline 1-Authentication/7-sign-in-express-mfa/App/auth/AuthProvider.js:78]: I don't think this is a pattern we want to demonstrate - parsing access tokens will not work when token encryption is enabled, rendering access tokens opaque. Instead, I think we implement the sample such that  the client doesn't need to know if the call requires MFA or not and just "does the right thing" when MFA is required.

File: 1-Authentication/7-sign-in-express-mfa/AppCreationScripts/sample.json [hunk 3]
```diff
@@ -0,0 +1,67 @@
+{
+    "Sample": {
+        "Title": "A Node.js & Express web app authenticating users against Azure AD for Customers with MSAL Node",
```
Comment (iambmelt) [inline 1-Authentication/7-sign-in-express-mfa/AppCreationScripts/sample.json:3]:
```suggestion
        "Title": "A Node.js & Express web app authenticating users against Entra ID for Customers with MSAL Node",
```

File: 1-Authentication/7-sign-in-express-mfa/AppCreationScripts/sample.json [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        ],
+        "Products": [
+            "azure-active-directory",
+            "msal-node"
+        ],
+        "RepositoryUrl": "ms-identity-ciam-javascript-tutorial",
+        "Endpoint": "AAD v2.0",
+        "Provider": "CIAM",
+        "Platform": "JavaScript",
+        "description": "This sample demonstrates a Node.js & Express web app authenticating users against Azure Active Directory Customer Identity Access Management (Azure AD for Customers) with Microsoft Authentication Library for Node (MSAL Node)"
```
Comment (iambmelt) [inline 1-Authentication/7-sign-in-express-mfa/AppCreationScripts/sample.json:17]:
```suggestion
        "description": "This sample demonstrates a Node.js & Express web app authenticating users with Microsoft Entra External ID using the Microsoft Authentication Library for Node (MSAL Node)"
```

**Review by iambmelt — CHANGES_REQUESTED** [review]
Some minor nit + 1 bigger nit on AT cracking

File: 1-Authentication/7-sign-in-express-mfa/App/views/id.hbs (hunk 1 above)
Comment (4gust) [reply 1-Authentication/7-sign-in-express-mfa/App/views/id.hbs]: Updated it

File: 1-Authentication/7-sign-in-express-mfa/App/auth/AuthProvider.js (hunk 2 above)
Comment (4gust) [reply 1-Authentication/7-sign-in-express-mfa/App/auth/AuthProvider.js]: Removed the code that did this.
Updated the flow you suggested.

File: 1-Authentication/7-sign-in-express-mfa/AppCreationScripts/sample.json (hunk 3 above)
Comment (4gust) [reply 1-Authentication/7-sign-in-express-mfa/AppCreationScripts/sample.json]: Updated the file

File: 1-Authentication/7-sign-in-express-mfa/App/package.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  "name": "msal-node-auth-code",
+  "version": "1.0.0",
+  "description": "sample web app for msal-node",
+  "scripts": {
+    "start": "node server.js"
+  },
+  "author": "Microsoft",
+  "license": "MIT",
+  "dependencies": {
+    "@azure/msal-node": "^2.8.0",
```
Comment (iambmelt) [inline 1-Authentication/7-sign-in-express-mfa/App/package.json:11]: Can we use the latest version here? Should be `2.9.2`

Comment (iambmelt) [reply 1-Authentication/7-sign-in-express-mfa/App/package.json:11]: Same for all the dependencies here

**Review by iambmelt — APPROVED** [review]
Plz update `msal-node` to latest but otherwise LGTM

**Review by iambmelt — APPROVED** [review]

**Pull request merged**